### PR TITLE
gnum4: 1.4.18 -> 1.4.19

### PR DESCRIPTION
**Pull request by KAction**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/gnum4/default.nix [hunk 1]
```diff
@@ -7,27 +7,19 @@
...
     url = "mirror://gnu/m4/m4-${version}.tar.bz2";
-    sha256 = "1xkwwq0sgv05cla0g0a01yzhk0wpsn9y40w9kh9miiiv0imxfh36";
+    sha256 = "sha256-swapHA/ZO8QoDPwumMt6s5gf91oYe+oyk4EfRSyJqMg=";
   };
 
   doCheck = false;
 
   configureFlags = [ "--with-syscmd-shell=${stdenv.shell}" ];
 
   # Upstream is aware of it; it may be in the next release.
```
Comment (alyssais) [inline pkgs/development/tools/misc/gnum4/default.nix:21]: Presumably this comment was referring to the patch that you removed?

Comment (KAction) [reply pkgs/development/tools/misc/gnum4/default.nix]: That is right. Fixed.

**Comment by veprbl** [comment]
Please target staging

**Comment by KAction** [comment]
Changed.

**Review by veprbl — APPROVED** [review]
Reviewed upstream changelog
https://lists.gnu.org/archive/html/m4-announce/2021-05/msg00002.html

**Pull request merged**